test/externalTests/exttest/common.py

```diff
@@ -0,0 +1,338 @@
+
+    print(f"Cloning {ref_type} {ref} of {repo_url}...")
+    if ref_type == "commit":
+        os.mkdir(test_dir)
+        os.chdir(test_dir)
+        git("init")
+        git(f"remote add origin {repo_url}")
+        git(f"fetch --depth 1 origin {ref}")
+        git("reset --hard FETCH_HEAD")
+    else:
+        os.chdir(test_dir.parent)
+        git(f"clone --depth 1 {repo_url} -b {ref} {test_dir.resolve()}")
+        if not test_dir.exists():
+            raise RuntimeError("Git clone failed.")
+        os.chdir(test_dir)
+
+    if (test_dir / ".gitmodules").exists():
+        git("submodule update --init")
+
+    print(f"Current commit hash: {git_commit_hash()}")
+
+
+def parse_solc_version(solc_version_string):
+    solc_version_match = re.search(SOLC_FULL_VERSION_REGEX, solc_version_string)
+    if solc_version_match:
+        return solc_version_match.group(1)
+    raise RuntimeError(
+        f"Solc version could not be found in: {solc_version_string}."
+    )
+
+
+def get_solc_short_version(solc_full_version):
+    solc_short_version_match = re.search(SOLC_SHORT_VERSION_REGEX, solc_full_version)
+    if solc_short_version_match:
+        return solc_short_version_match.group(1)
+    raise RuntimeError(
+        f"Error extracting short version string from: {solc_full_version}."
+    )
+
+
+def setup_solc(config: TestConfig, test_dir: Path) -> (str, str):
+    sc_config = config.solc
+
+    if sc_config.binary_type == "solcjs":
+        solc_dir = test_dir.parent / sc_config.install_dir
+        solc_bin = solc_dir / "dist/solc.js"
+
+        print("Setting up solc-js...")
+        if sc_config.solcjs_src_dir == "":
+            download_project(
+                solc_dir,
+                "https://github.com/ethereum/solc-js.git",
+                "branch",
+                sc_config.branch,
+            )
+        else:
+            print(f"Using local solc-js from {sc_config.solcjs_src_dir}...")
+            copytree(sc_config.solcjs_src_dir, solc_dir)
+            rmtree(solc_dir / "dist")
+            rmtree(solc_dir / "node_modules")
+        os.chdir(solc_dir)
+        run_cmd("npm install")
+        run_cmd("npm run build")
+
+        if mimetypes.guess_type(sc_config.binary_path)[0] != "application/javascript":
+            raise WrongBinaryType(
+                "Provided soljson.js is expected to be of the type application/javascript but it is not."
+            )
+
+        copyfile(sc_config.binary_path, solc_dir / "dist/soljson.js")
+        solc_version_output = subprocess.getoutput(f"node {solc_bin} --version")
+    else:
+        print("Setting up solc...")
+        solc_version_output = subprocess.getoutput(
+            f"{sc_config.binary_path} --version"
+        ).split(":")[1]
+
+    return parse_solc_version(solc_version_output)
+
+
+def store_benchmark_report(self):
+    # TODO
+    raise NotImplementedError()
+
+
+def prepare_node_env(test_dir: Path):
```

Comment:
This is not relevant for the foundry tests itself, but is part of the migration to the new python-based external tests suite.

Reply:
I think it would be better to split them off because in this PR this code is not getting executed so we can't even be sure it will work. I'd rather have it included in a PR that converts at least one test using it.

Same with support for `soljson.js`, which I assume will not work with Foundry based projects.